execution-engine/engine-grpc-server/src/engine_server/mappings/ipc/public_key.rs

```diff
@@ -0,0 +1,66 @@
+use std::convert::TryFrom;
+
+use types::account::PublicKey;
+
+use crate::engine_server::{mappings::MappingError, state};
+
+impl From<PublicKey> for state::PublicKey {
+    fn from(public_key: PublicKey) -> Self {
+        let raw_bytes = {
+            let PublicKey::Ed25519(ed25519) = public_key;
+            ed25519.value().to_vec()
+        };
+
+        let mut result = state::PublicKey::new();
+        let mut ed25519 = state::Ed25519::new();
+        ed25519.set_public_key(raw_bytes);
+        result.set_ed25519(ed25519);
+        result
+    }
+}
+
+impl TryFrom<state::PublicKey> for PublicKey {
+    type Error = MappingError;
+    fn try_from(public_key: state::PublicKey) -> Result<Self, Self::Error> {
+        if !public_key.has_ed25519() {
+            return Err(MappingError::MissingPayload);
+        }
+        let ed25519 = public_key.get_ed25519();
+        let raw_bytes = ed25519.get_public_key();
+        PublicKey::ed25519_try_from(&raw_bytes)
+            .map_err(|_| MappingError::invalid_public_key_length(raw_bytes.len()))
+    }
```

Comment:
```suggestion

    fn try_from(pb_public_key: state::PublicKey) -> Result<Self, Self::Error> {
        let pb_public_key = pb_public_key
            .variant
            .ok_or_else(|| ParsingError("Unable to parse Protobuf PublicKey".to_string()))?;

        let public_key = match pb_public_key {
            PublicKey_oneof_variant::ed25519(ed25519) => {
                let raw_bytes = ed25519.get_public_key();
                PublicKey::ed25519_try_from(&raw_bytes)
                    .map_err(|_| MappingError::invalid_public_key_length(raw_bytes.len()))?
            }
        };

        Ok(public_key)
    }
```
This is more in line with how we handle other mappings to enums, and sets us up for handling further PublicKey variants.

Reply:
Good point with the enum. Oh and how did I miss `ParsingError` nested inside `MappingError`

Reply:
Done